src/itinerary/Itinerary.tsx

```diff
@@ -78,132 +131,26 @@ const Itinerary = ({
-                {
-                  'kirk-itinerary--arrival': isLastPlace,
-                  'kirk-itinerary-location--withToAddon': isLastPlace && isNonEmptyString(toAddon),
-                },
-              ])}
-              key={computeKeyFromPlace(place)}
-              itemProp="location"
-              itemScope
-              itemType="http://schema.org/Place"
-              aria-label={place.stepAriaLabel}
-            >
-              <meta itemProp="name" content={place.mainLabel} />
-              <meta
-                itemProp="address"
-                content={
-                  place.subLabel && typeof place.subLabel === 'string'
-                    ? place.subLabel
-                    : place.mainLabel
-                }
-              />
-              <Component {...hrefProps} aria-label={place.actionAriaLabel}>
-                {hasTime && (
-                  <time dateTime={place.isoDate}>
-                    <Text tag={TextTagType.DIV} display={TextDisplayType.TITLESTRONG}>
-                      {place.time}
-                    </Text>
-                  </time>
-                )}
-                <div className="kirk-itinerary-location-city">
-                  <div className="kirk-itinerary-road" aria-hidden="true" />
-                  <Bullet />
-                  <Text tag={TextTagType.PARAGRAPH} display={TextDisplayType.TITLESTRONG}>
-                    {place.mainLabel}
-                  </Text>
-                  {hasSubLabel &&
-                    (typeof place.subLabel === 'string' ? (
-                      <Text
-                        tag={TextTagType.PARAGRAPH}
-                        display={TextDisplayType.CAPTION}
-                        textColor={color.primaryText}
-                      >
-                        {place.subLabel}
-                      </Text>
-                    ) : (
-                      <div>{place.subLabel}</div>
-                    ))}
-                </div>
-                {hasChevron && (
-                  <div className="kirk-itinerary-location-chevron">
-                    <ChevronIcon />
-                  </div>
-                )}
-              </Component>
-            </li>
-          )
-        })}
-        {isNonEmptyString(toAddon) && (
-          <li
-            className="kirk-itinerary-location kirk-itinerary-addon kirk-itinerary-addon--to"
-            aria-label={toAddonAriaLabel}
-          >
-            <Bullet type={BulletTypes.ADDON} />
-            <div className="kirk-itinerary-location-city">
-              <Text
-                tag={TextTagType.PARAGRAPH}
-                display={TextDisplayType.CAPTION}
-                textColor={color.fadedText}
-              >
-                {toAddon}
-              </Text>
-            </div>
-          </li>
+        {renderAddon('from', fromAddon, fromAddonAriaLabel)}
+        {renderDepartureOrArrival(places, false, small, withTime, false)}
```

Comment:
Not a fan of this naming, it's kind of misleading: we render an addon (from), followed by a departure OR an arrival. IMO we should directly call a `renderLocation` with the right place as a parameter.